src/engine/strat_engine/backstore/metadata/bda.rs

```diff
@@ -346,84 +402,34 @@ impl StaticHeader {
                         } else if loc_1.initialization_time > loc_2.initialization_time {
                             // If the first header block is newer, overwrite second with
                             // contents of first.
-                            loc_1.write(f, MetadataLocation::Second)?;
-                            Ok(Some(loc_1))
+                            write_header(loc_1, MetadataLocation::Second, f)
                         } else {
                             // The second header block must be newer, so overwrite first
                             // with contents of second.
-                            loc_2.write(f, MetadataLocation::First)?;
-                            Ok(Some(loc_2))
+                            write_header(loc_2, MetadataLocation::First, f)
                         }
                     }
                     (None, None) => Ok(None),
-                    (Some(loc_1), None) => {
-                        // Copy 1 has valid Stratis BDA, copy 2 has no magic, re-write copy 2
-                        loc_1.write(f, MetadataLocation::Second)?;
-                        Ok(Some(loc_1))
-                    }
-                    (None, Some(loc_2)) => {
-                        // Copy 2 has valid Stratis BDA, copy 1 has no magic, re-write copy 1
-                        loc_2.write(f, MetadataLocation::First)?;
-                        Ok(Some(loc_2))
-                    }
+                    // Copy 1 has valid Stratis BDA, copy 2 has no magic, re-write copy 2
```

Comment:
Also, could you remove the two comments about the valid Stratis BDA? Only the sigblock has been read, so it is not certain that the BDA is valid, so these two comments are misleading.